browser/brave_shields/brave_shields_web_contents_observer.cc

```diff
@@ -84,44 +85,36 @@ void UpdateContentSettingsToRendererFrames(content::WebContents* web_contents) {
 
 namespace brave_shields {
 
-base::Lock BraveShieldsWebContentsObserver::frame_data_map_lock_;
+base::NoDestructor<std::map<int, GURL>> frame_tree_node_id_to_tab_url_;
```

Comment:
we really shouldn't need this at all since it's all on the UI thread now, but it's outside the scope of this PR